mitodl/mit-open
# Departments Listing Page

### What are the relevant tickets?
- https://github.com/mitodl/hq/issues/4108
- Figma: https://www.figma.com/file/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?type=design&node-id=1215-19081&mode=design&t=8D9Lk5k0p0HaTrln-0

### Description (What does it do?)
This PR adds a department listing page.

### Screenshots (if appropriate):

See https://github.com/mitodl/mit-open/pull/865#issuecomment-2096829131

<details>
<summary>Show original screenshots.</summary>
  
<img width="1000" alt="Screenshot 2024-05-03 at 3 54 25 PM" src="https://github.com/mitodl/mit-open/assets/9010790/4cbe4bca-381e-4791-95b1-2cfba6204ac3">

<img width="375" alt="Screenshot 2024-05-03 at 3 56 08 PM" src="https://github.com/mitodl/mit-open/assets/9010790/a75487e1-707f-497c-a263-cae11805124e">

</details>


### How can this be tested?

1. Ensure that you have populated channels: `./manage.py backpopulate_resource_channels`. This creates a "channel" for each department. The department listing page links to the "channel pages".
2. On this branch, `docker compose build watch` (a new package was added, and they don't get installed on `up` any more). Then `docker compose up`.
3. View http://localhost:8063/departments. 
    - It should look like the figma designs, and you should
    - you should be able to click on departments to go to the department details page (which are WIPs).

### Additional Context
